From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pekka Vuorela <pekka.vuorela@jolla.com>
Date: Wed, 6 Apr 2022 17:56:07 +0300
Subject: [PATCH] Install unit test files

Before switch to cmake on commit 7471153ce the unit test files were
installed, and with cmake there are some parts of it done.
Reintroduce the installation as these are needed if the build
environment is not, and perhap cannot be, used for testing.
---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7d4d1f2..eecc1aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -490,6 +490,7 @@ if(enable-tests)
     set(DUMMY_PLUGINS dummyimplugin dummyimplugin2 dummyimplugin3 dummyplugin)
 
     set_property(TARGET ${DUMMY_PLUGINS} PROPERTY LIBRARY_OUTPUT_DIRECTORY ${TEST_PLUGINS_DIR})
+    install(TARGETS ${DUMMY_PLUGINS} DESTINATION ${CMAKE_INSTALL_LIBDIR}/maliit-framework-tests/plugins)
 
     add_library(test-utils STATIC
             tests/utils/core-utils.cpp
@@ -519,6 +520,7 @@ if(enable-tests)
         list(REMOVE_AT _libs 0)
         target_link_libraries(${name} test-utils maliit-plugins ${_libs})
         add_test(${name} ${name})
+        install(TARGETS ${name} DESTINATION ${CMAKE_INSTALL_LIBDIR}/maliit-framework-tests/${name})
 
         set(test_targets ${test_targets} ${name} PARENT_SCOPE)
         if(install-tests)
@@ -528,8 +530,14 @@ if(enable-tests)
 
     create_test(sanitychecks)
     create_test(ut_mattributeextensionmanager)
+    install(DIRECTORY tests/ut_mattributeextensionmanager/
+           DESTINATION ${CMAKE_INSTALL_LIBDIR}/maliit-framework-tests/ut_mattributeextensionmanager
+           FILES_MATCHING PATTERN "*.xml")
     create_test(ut_mimonscreenplugins)
     create_test(ut_mimpluginmanager ${DUMMY_PLUGINS})
+    install(DIRECTORY tests/ut_mimpluginmanager/
+           DESTINATION ${CMAKE_INSTALL_LIBDIR}/maliit-framework-tests/ut_mimpluginmanager
+           FILES_MATCHING PATTERN "*.xml")
     create_test(ut_mimpluginmanagerconfig)
     create_test(ut_mimserveroptions)
     create_test(ut_mimsettings)
